### ISSUE TITLE
Question about SSIM evaluaton, training epoch and the input of content fusion module

### ISSUE THREAD

**Issue by LeoXing1996** [switchablenorms/DeepFashion_Try_On#14]
Thanks for this great work, however, I still have some questions about this work.

1. How should I evaluate SSIM? I directly calculate SSIM score on image reconstruction task (reference and target are from the same image). However, the pretrained model and my own trained model (20 epoch) get 0.7980 and 0.7594 on test set.

2. Does this model only need 20 epochs training? In default options, this model would be trained for 200 epochs. I found that SSIM score is still increasing after 20 epochs and reached 0.7653 in epoch 40.

3. In released training and inference code, average skin color (`skin_color`) of each class area is used in the input of content fusion module instead of synthesized clothing mask (M_c^S) mentioned in paper. 🤔
https://github.com/switchablenorms/DeepFashion_Try_On/blob/a628ca5612431ab5e8c0e729265e9d3453d71575/ACGPN_train/models/pix2pixHD_model.py#L323

**Comment by Amazingren** [comment]
Hi @LeoXing1996, I am also want to replicate the SSIM metric for evaluation. However, I have no idea about how to do this by calculating SSIM between generated Images and ground truth images.
So how should I find the ground truth images firstly? It would be great help if you can share some advice with me on this issue~

**Comment by LeoXing1996** [comment]
In SSIM evaluation, the model performs a reconstruction task, like the training process.
Therefore, GT images are from the test set.

**Comment by LeoXing1996** [comment]
> Hi @LeoXing1996, I am also want to replicate the SSIM metric for evaluation. However, I have no idea about how to do this by calculating SSIM between generated Images and ground truth images.
> So how should I find the ground truth images firstly? It would be great help if you can share some advice with me on this issue~

In SSIM evaluation, the model performs a reconstruction task, like the training process.
Therefore, GT images are from the test set.
